lib/docx/document.rb

```diff
@@ -198,7 +210,9 @@ def update
 
     # generate Elements::Containers::Paragraph from paragraph XML node
     def parse_paragraph_from(p_node)
-      Elements::Containers::Paragraph.new(p_node, document_properties)
+      p = Elements::Containers::Paragraph.new(p_node, document_properties)
+      p.document = self
+      p
```

Comment:
`#document=` looks much open: anybody can replace another `Document` instance with this accessor of the paragraph returned.
how about giving Document object to Paragraph's constructor like `Element::Containers::Paragraph.new(p_node, document_properties, self)` ?

Reply:
yes it's ok, my initial idea was to avoid modifying previous API.
I modify branch to take into account.